Kiosk/Admin/AdminCardTestingViewController.swift

```diff
@@ -21,27 +21,28 @@ class AdminCardTestingViewController: UIViewController {
             cardHandler = CardHandler(apiKey: self.keys.cardflightProductionAPIClientKey(), accountToken: self.keys.cardflightProductionMerchantAccountToken())
         }
 
-        cardHandler.cardSwipedSignal.subscribeNext({ (message) -> Void in
-                self.log("\(message)")
-                return
-
-            }, error: { (error) -> Void in
-
-                self.log("\n====Error====\n\(error)\nThe card reader may have become disconnected.\n\n")
-                if self.cardHandler.card != nil {
-                    self.log("==\n\(self.cardHandler.card!)\n\n")
-                }
-
-
-            }, completed: {
-
-                if let card = self.cardHandler.card {
-                    let cardDetails = "Card: \(card.name) - \(card.last4) \n \(card.cardToken)"
-                    self.log(cardDetails)
+        cardHandler.cardStatus
+            .subscribe { (event) -> Void in
+                switch event {
+                case .Next(let message):
+                    self.log("\(message)")
+                case .Error(let error):
+                    self.log("\n====Error====\n\(error)\nThe card reader may have become disconnected.\n\n")
+                    if self.cardHandler.card != nil {
+                        self.log("==\n\(self.cardHandler.card!)\n\n")
+                    }
+                case .Completed:
+                    if let card = self.cardHandler.card {
+                        let cardDetails = "Card: \(card.name) - \(card.last4) \n \(card.cardToken)"
+                        self.log(cardDetails)
```

Comment:
and I bet that `let string = thing` then logging was an Xcode/swift bug that may not exist anymore